test/e2e/monitoring_stack_controller_test.go

```diff
@@ -122,6 +128,23 @@ func emptyStackCreatesPrometheus(t *testing.T) {
 	f.GetResourceWithRetry(t, ms.Name, ms.Namespace, &prometheus)
 }
 
+func nilResrouceSelectorPropagatesToPrometheus(t *testing.T) {
+	ms := newMonitoringStack(t, "nil-selector")
+	err := f.K8sClient.Create(context.Background(), ms)
+	assert.NilError(t, err, "failed to create a monitoring stack")
+	f.AssertResourceEventuallyExists(ms.Name, ms.Namespace, &monv1.Prometheus{})(t)
+
+	updatedMS := &stack.MonitoringStack{}
+	f.GetResourceWithRetry(t, ms.Name, ms.Namespace, updatedMS)
+	updatedMS.Spec.ResourceSelector = nil
+	err = f.K8sClient.Update(context.Background(), updatedMS)
+	assert.NilError(t, err, "failed to patch monitoring stack with nil resource selector")
+
+	prometheus := monv1.Prometheus{}
+	f.GetResourceWithRetry(t, updatedMS.Name, updatedMS.Namespace, &prometheus)
```

Comment:
this may yield failure depending on the timing because `GetResourceWithRetry` may give us the old Prometheus ... we need to assert the generation has changed whenever we modify an exisiting stack.